workflowparsers/lobster/parser.py

```diff
@@ -67,96 +68,215 @@ def icoxp_line_split(string):
                 [int(tmp[4]), int(tmp[5]), int(tmp[6])],
                 float(tmp[7]),
             ]
+        elif len(tmp) == 9:
+            return [
+                tmp[1],
+                tmp[2],
+                float(tmp[3]),
+                [int(tmp[4]), int(tmp[5]), int(tmp[6])],
+                float(tmp[7]), float(tmp[8]),
+            ]
         # LOBSTER versions below 3
-        elif len(tmp) == 6:
+        elif len(tmp) == 6 and tmp[-2] != "spin":
             return [tmp[1], tmp[2], float(tmp[3]), float(tmp[4]), int(tmp[5])]
 
-    icoxplist_parser = TextParser(
-        quantities=[
-            Quantity(
-                'icoxpslist_for_spin',
-                r'\s*CO[OH]P.*spin\s*\d\s*([^#]+[-\d\.]+)',
-                repeats=True,
-                sub_parser=TextParser(
-                    quantities=[
-                        Quantity(
-                            'line',
-                            # LOBSTER version 3 and above
-                            r'(\s*\d+\s+\w+\s+\w+\s+[\.\d]+\s+[-\d]+\s+[-\d]+\s+[-\d]+\s+[-\.\d]+\s*)|'
-                            # LOBSTER versions below 3
-                            r'(\s*\d+\s+\w+\s+\w+\s+[\.\d]+\s+[-\.\d]+\s+[\d]+\s*)',
-                            repeats=True,
-                            str_operation=icoxp_line_split,
-                        )
-                    ]
-                ),
-            )
-        ]
-    )
+    float_version = float(version.split(".")[0]+"."+version.split(".")[1])
+    if 5 > float_version > 2:
+        icoxplist_parser = TextParser(
+            quantities=[
+                Quantity(
+                    "icoxpslist_for_spin",
+                    r"\s*(CO[O,H,B,I,P]).*spin\s*\d\s*([^#]+[-\d\.]+)",
+                    repeats=True,
+                    sub_parser=TextParser(
+                        quantities=[
+                            Quantity(
+                                "line",
+                                # LOBSTER version 3 and above
+                                r"(\s*\d+\s+\w+\s+\w+\s+[\.\d]+\s+[-\d]+\s+[-\d]+\s+[-\d]+\s+[-\.\d]+\s*)",
```

Comment:
if you want, to make it a bit more readable you can directly use a space instead of \s, and also, do we need + or * when referring to the spaces between the information?

The resulting would look like:
```python
                                r"( *\d+ *\w+ *\w+ *[\.\d]+ *[-\d]+ *[-\d]+ *[-\d]+ *[-\.\d]+)",
```

Reply:
Done